app/index.js

```diff
@@ -99,147 +155,230 @@ var AspnetGenerator = yeoman.generators.Base.extend({
       message: 'What\'s the name of your ASP.NET application?',
       default: app
     }];
+    this.prompt(prompts, cb);
+  },
+
+  askForProjectStructure: function() {
+    var done = this.async();
+
+    var setApplicationDirectory = function() {
+      if (this.type === 'unittest') {
+        this.applicationDirectory = 'tests/' + this.applicationName;
+      } else {
+        this.applicationDirectory = 'src/' + this.applicationName;
+      }
+    }.bind(this);
+
+    if (this.options.composing) {
+      this.projectStructure = true;
+      setApplicationDirectory();
+      done();
+      return;
+    }
+
+    if (config.getGlobalJsonPath()) {
+      var globalJsonDirectory = path.resolve(path.dirname(config.getGlobalJsonPath()));
+      if (fs.existsSync(path.join(globalJsonDirectory, 'tests')) || fs.existsSync(path.join(globalJsonDirectory, 'src'))) {
+        setApplicationDirectory();
+        if (this.destinationRoot() !== globalJsonDirectory) {
+          this.destinationRoot(globalJsonDirectory);
+        }
+      } else {
+        this.applicationDirectory = this.applicationName;
+      }
+
+      done();
+      return;
+    }
+
+    var prompts = [{
+      type: 'confirm',
+      name: 'projectStructure',
+      message: 'Create a solution file?',
+      default: true
```

Comment:
Honestly doesn't matter to me, the naming of "file" probably needs to change.  Maybe it should just be `Create global.json file?`